src/Guards/TokenGuard.php

```diff
@@ -82,6 +82,17 @@ public function __construct(ResourceServer $server,
         $this->encrypter = $encrypter;
     }
 
+    /**
+     * Determine if the requested provider matches the client's provider.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return bool
+     */
+    protected function validateProvider(Request $request)
+    {
+        return $this->provider == $this->client($request)->getProvider();
```

Comment:
Best to use strict comparison here.

```suggestion
        return $this->provider === $this->client($request)->getProvider();
```

Reply:
Strict comparison won't work since we may be comparing different instances of the same object.

Reply:
@billriess aren't we comparing strings here?

Reply:
No, we're comparing implementations of Illuminate\Contracts\Auth\UserProvider.